src/asmcnc/apps/drywall_cutter_app/screen_config_filechooser.py

```diff
@@ -0,0 +1,524 @@
+
+
+""")
+
+configs_dir = './asmcnc/apps/drywall_cutter_app/config/configurations/'  # where job files are cached for selection (for last used history/easy access)
+
+
+def date_order_sort(files, filesystem):
+    return (sorted(f for f in files if filesystem.is_dir(f)) +
+            sorted((f for f in files if not filesystem.is_dir(f)), key=lambda fi: os.stat(fi).st_mtime, reverse=False))
+
+
+def date_order_sort_reverse(files, filesystem):
+    return (sorted(f for f in files if filesystem.is_dir(f)) +
+            sorted((f for f in files if not filesystem.is_dir(f)), key=lambda fi: os.stat(fi).st_mtime, reverse=True))
+
+
+def name_order_sort(files, filesystem):
+    return (sorted(f for f in files if filesystem.is_dir(f)) +
+            sorted(f for f in files if not filesystem.is_dir(f)))
+
+
+def name_order_sort_reverse(files, filesystem):
+    return (sorted(f for f in files if filesystem.is_dir(f)) +
+            sorted((f for f in files if not filesystem.is_dir(f)), reverse=True))
+
+
+decode_and_encode = lambda x: (unicode(x, detect(x)['encoding'] or 'utf-8').encode('utf-8'))
+
+
+class ConfigFileChooser(Screen):
+    filename_selected_label_text = StringProperty()
+
+    sort_by_date = ObjectProperty(date_order_sort)
+    sort_by_date_reverse = ObjectProperty(date_order_sort_reverse)
+    sort_by_name = ObjectProperty(name_order_sort)
+    sort_by_name_reverse = ObjectProperty(name_order_sort_reverse)
+    is_filechooser_scrolling = False
+
+    def __init__(self, **kwargs):
+        super(ConfigFileChooser, self).__init__(**kwargs)
+        self.sm = kwargs['screen_manager']
+        self.l = kwargs['localization']
+        self.callback = kwargs['callback']
+        self.usb_stick = usb_storage.USB_storage(self.sm,
+                                                 self.l)  # object to manage presence of USB stick (fun in Linux)
+
+        self.check_for_job_cache_dir()
+
+        # MANAGING KIVY SCROLL BUG
+
+        self.list_layout_fc.ids.scrollview.bind(on_scroll_stop=self.scrolling_stop)
+        self.list_layout_fc.ids.scrollview.bind(on_scroll_start=self.scrolling_start)
+        self.icon_layout_fc.ids.scrollview.bind(on_scroll_stop=self.scrolling_stop)
+        self.icon_layout_fc.ids.scrollview.bind(on_scroll_start=self.scrolling_start)
+
+        self.list_layout_fc.ids.scrollview.effect_cls = kivy.effects.scroll.ScrollEffect
+        self.icon_layout_fc.ids.scrollview.effect_cls = kivy.effects.scroll.ScrollEffect
+
+        self.icon_layout_fc.ids.scrollview.funbind('scroll_y', self.icon_layout_fc.ids.scrollview._update_effect_bounds)
+        self.list_layout_fc.ids.scrollview.funbind('scroll_y', self.list_layout_fc.ids.scrollview._update_effect_bounds)
+        self.icon_layout_fc.ids.scrollview.fbind('scroll_y', self.alternate_update_effect_bounds_icon)
+        self.list_layout_fc.ids.scrollview.fbind('scroll_y', self.alternate_update_effect_bounds_list)
+
+        self.json_config_order = {
```

Comment:
Move this out of the kivy scroll bug stuff - it's a bit jarring and confusing here!